pkg/controller/statspuller/signer.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statspuller
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+)
+
+type SignerInfo struct {
+	Signer crypto.Signer
+	KeyID  string
+	Issuer string
+}
+
+func (c *Controller) getSignerForRealm(ctx context.Context, realmID uint) (*SignerInfo, error) {
```

Comment:
I think we need to optimize this function a bit and consume the public key cache.